src/lightning_app/storage/drive.py

```diff
@@ -231,6 +231,8 @@ def __deepcopy__(self, memo):
     def _collect_component_names(self) -> List[str]:
         sep = "/"
         if self.fs.exists(self.drive_root):
+            # Invalidate cache before running ls in case new directories have been added
+            self.fs.invalidate_cache()
```

Comment:
do we know if this has any performance implications?

Reply:
For sure this has implications with large nested fs layouts (which we don't have atm but they may come in the future, especially as we roll out training features). We don't have a workaround for now - the only one being monitoring changes incrementally (everytime we do an operation through the Drive API we update the cache accordingly).

@ethanwharris I would add a TODO here to re-evaluate.